sdk/synapse/synapse/src/models/index.ts

```diff
@@ -0,0 +1,970 @@
+  sparkApplicationId?: string;
+  livyId?: string;
+  timing?: string[];
+  sparkJobDefinition?: string;
+  pipelineProperty?: SparkJob[];
+  jobType?: string;
+  submitTime?: Date;
+  endTime?: Date;
+  queuedDuration?: string;
+  runningDuration?: string;
+  totalDuration?: string;
+}
+
+/**
+ * An interface representing SparkJobListViewResponse.
+ */
+export interface SparkJobListViewResponse {
+  nJobs?: number;
+  sparkJobs?: SparkJob[];
+}
+
+/**
+ * An interface representing HistoryServerPropertiesResponse.
+ */
+export interface HistoryServerPropertiesResponse {
+  webProxyEndpoint?: string;
+}
+
+/**
+ * An interface representing Option.
+ */
+export interface Option {
+  additionalProp1?: string;
+  additionalProp2?: string;
+  additionalProp3?: string;
+}
+
+/**
+ * An interface representing Tables.
+ */
+export interface Tables {
+  operation?: string;
+  options?: Option[];
+}
+
+/**
+ * An interface representing Config.
+ */
+export interface Config {
+  isHiveEnabled?: boolean;
+}
+
+/**
+ * An interface representing Data.
+ */
+export interface Data {
+  handlerPath?: string;
+  inputs?: string[];
+  outputs?: string[];
+  tables?: Tables;
+  config?: Config;
+}
+
+/**
+ * An interface representing HistoryServerDataResponse.
+ */
+export interface HistoryServerDataResponse {
+  data?: Data;
+}
+
+/**
+ * An interface representing Stages.
+ */
+export interface Stages {
+  dataAvg?: number;
+  timeAvg?: number;
+  id?: number;
+  attemptId?: number;
+  name?: string;
+  dataSkewTask?: number[];
+  timeSkewTask?: number[];
+  tasks?: number[];
+}
+
+/**
+ * An interface representing Executors.
+ */
+export interface Executors {
+  isDynamicAllocationOn?: boolean;
+  executorInstances?: number;
+  startTime?: number;
+  endTime?: number;
+  events?: number[];
+  executorCores?: string;
+  executorMemory?: string;
+}
+
+/**
+ * An interface representing HistoryServerDiagnosticResponseData.
+ */
+export interface HistoryServerDiagnosticResponseData {
+  stages?: Stages;
+  executors?: Executors;
+}
+
+/**
+ * An interface representing HistoryServerDiagnosticResponse.
+ */
+export interface HistoryServerDiagnosticResponse {
+  data?: HistoryServerDiagnosticResponseData;
+}
+
+/**
+ * An interface representing Edge.
+ */
+export interface Edge {
+  id?: number;
+  childId?: number;
+}
+
+/**
+ * An interface representing Jobs.
+ */
+export interface Jobs {
+  time?: number;
+  startTime?: number;
+  id?: number;
+  edges?: Edge[];
+  nodes?: number[];
+}
+
+/**
+ * An interface representing HistoryServerGraphResponseData.
+ */
+export interface HistoryServerGraphResponseData {
+  isAppFinished?: boolean;
+  jobs?: Jobs;
+  stages?: Stages;
+}
+
+/**
+ * An interface representing HistoryServerGraphResponse.
+ */
+export interface HistoryServerGraphResponse {
+  data?: HistoryServerGraphResponseData;
+}
+
+/**
+ * An interface representing LivyRequestBase.
+ */
+export interface LivyRequestBase {
+  name?: string;
+  file?: string;
```

Comment:
> Can `file` and `files` both be returned by the service? Do they represent different things?

Hi this is livy standard. We can't modify this. They will be returned by the service

Reply:
Got it, thanks for the clarification!

Reply:
Can you link the relevant standard? Often a simple translation of spec structures to TS doesn't take advantage of TS's extremely advanced type system. E.g. consider a case where an object either has a and b, or c and d properties. A naive approach is to type this `{a?: number, b?: number, c?: number, d?: number}` since in practice any field might be undefined. However, `{a: number, b: number } | {c: number, d: number}` is more accurate, typechecks more tightly, and gives a better tooling experience.